include/highfive/bits/H5Utils.hpp

```diff
@@ -68,6 +76,15 @@ struct inspector<std::vector<T>> {
 
     static constexpr size_t number_dimensions = 1;
     static constexpr size_t recursive_number_dimensions = number_dimensions + inspector<value_type>::recursive_number_dimensions;
+
+    static std::array<size_t, recursive_number_dimensions> getDimensions(const type& val) {
```

Comment:
But then here I am confused, I thought I understood well but I'm starting to develop doubts... Isn't this `shape`??

Reply:
Yes, but we call it `getDimensions`:
https://github.com/BlueBrain/HighFive/blob/master/include/highfive/H5DataSet.hpp#L87-L94

It's written here `getDimensions return shape`

Reply:
OK. Not a fan ;) I would say an nd-array has:

* `shape` : size along each axis
* `size == prod(shape)`
* `dimension == rank == ndim == shape.size()` (here names do vary a bit in different libraries)